tracing-journald fix for #1698 fails to build on glibc < 2.27

## Bug Report
Building tracing-journald 0.2.2 on Ubuntu Xenial (supported until 2024), Debian Stretch (supported until June 2022) or RHEL 7 (supported until 2024) fails with ``in function `tracing_journald::memfd::create': ... github.com-1ecc6299db9ec823/tracing-journald-0.2.2/src/memfd.rs:11: undefined reference to `memfd_create'``.

This syscall was introduced in Linux 3.17 (in 2014, safely out of LTS), but only exposed by glibc in 2.27 (released in Feb 2018). Unfortunately, `libc` (used by #1744) just calls out to glibc for this Linux extension, instead of directly invoking a `syscall(2)` stub, which would always link and return `ENOSYS` on very old kernels.

### Version

```
$ cargo tree | grep tracing
│   │   │   │   └── tracing v0.1.29
│   │   │   │       ├── tracing-attributes v0.1.18 (proc-macro)
│   │   │   │       └── tracing-core v0.1.21
│   │   │   │   ├── tracing v0.1.29 (*)
│   ├── tracing v0.1.29 (*)
│   ├── tracing-subscriber v0.3.7
│   │   ├── tracing v0.1.29 (*)
│   │   ├── tracing-core v0.1.21 (*)
│   │   └── tracing-log v0.1.2
│   │       └── tracing-core v0.1.21 (*)
│   └── tracing v0.1.29 (*)
├── tracing v0.1.29 (*)
├── tracing-journald v0.2.2
│   ├── tracing-core v0.1.21 (*)
│   └── tracing-subscriber v0.3.7 (*)
├── tracing-subscriber v0.3.7 (*)
    ├── tracing v0.1.29 (*)
```

### Platform

Ubuntu Xenial, Debian Stretch, RHEL 7, on x86-64.

### Crates

tracing-journald >= 0.2.1

### Description

I try to build a crate that depends on `tracing-journald`. Linking fails because no one defines `memfd_create`. Pinning to `0.2.0` restores the build.

--- a/tracing-journald/src/memfd.rs
+++ b/tracing-journald/src/memfd.rs
@@ -8,7 +8,13 @@ use std::os::raw::c_uint;
 use std::os::unix::prelude::{FromRawFd, RawFd};
 
 fn create(flags: c_uint) -> Result<File> {
-    let fd = unsafe { memfd_create("tracing-journald\0".as_ptr() as *const c_char, flags) };
+    let fd = unsafe {
+        syscall(
+            SYS_memfd_create,
+            "tracing-journald\0".as_ptr() as *const c_char,
+            flags,
+        )
+    };
     if fd < 0 {
         Err(Error::last_os_error())
     } else {

--- a/tracing-journald/src/memfd.rs
+++ b/tracing-journald/src/memfd.rs
@@ -8,6 +8,11 @@ use std::os::raw::c_uint;
 use std::os::unix::prelude::{FromRawFd, RawFd};
 
 fn create(flags: c_uint) -> Result<File> {
+    // Make the `memfd_create` syscall ourself instead of going through `libc`;
+    // `memfd_create` isn't supported on `glibc<2.27` so this allows us to
+    // support old-but-still-used distros like Ubuntu Xenial, Debian Stretch,
+    // RHEL 7, etc.
+    // See: https://github.com/tokio-rs/tracing/issues/1879
     let fd = unsafe {
         syscall(
             SYS_memfd_create,

--- a/tracing-journald/src/memfd.rs
+++ b/tracing-journald/src/memfd.rs
@@ -8,22 +8,27 @@ use std::os::raw::c_uint;
 use std::os::unix::prelude::{FromRawFd, RawFd};
 
 fn create(flags: c_uint) -> Result<File> {
-    // Make the `memfd_create` syscall ourself instead of going through `libc`;
-    // `memfd_create` isn't supported on `glibc<2.27` so this allows us to
-    // support old-but-still-used distros like Ubuntu Xenial, Debian Stretch,
-    // RHEL 7, etc.
-    // See: https://github.com/tokio-rs/tracing/issues/1879
-    let fd = unsafe {
+    let fd = memfd_create_syscall(flags);
+    if fd < 0 {
+        Err(Error::last_os_error())
+    } else {
+        Ok(unsafe { File::from_raw_fd(fd as RawFd) })
+    }
+}
+
+/// Make the `memfd_create` syscall ourself instead of going through `libc`;
+/// `memfd_create` isn't supported on `glibc<2.27` so this allows us to
+/// support old-but-still-used distros like Ubuntu Xenial, Debian Stretch,
+/// RHEL 7, etc.
+///
+/// See: https://github.com/tokio-rs/tracing/issues/1879
+fn memfd_create_syscall(flags: c_uint) -> i64 {
+    unsafe {
         syscall(
             SYS_memfd_create,
             "tracing-journald\0".as_ptr() as *const c_char,
             flags,
         )
-    };
-    if fd < 0 {
-        Err(Error::last_os_error())
-    } else {
-        Ok(unsafe { File::from_raw_fd(fd as RawFd) })
     }
 }
 

